src/components/nav_drawer/nav_drawer.tsx

```diff
@@ -140,7 +140,7 @@ export class EuiNavDrawer extends Component<
     this.setState({
       isLocked: !isLocked,
       isCollapsed: false,
-      outsideClickDisabled: true,
+      outsideClickDisabled: false,
```

Comment:
```suggestion
      outsideClickDisabled: !isLocked,
```

This still fixes the issue, but prevents the outside click callback from firing when locked

Reply:
Aah yes, perfect - I've updated the PR with a new squashed commit.